activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeConnectorTest.java

```diff
@@ -0,0 +1,55 @@
+package org.activiti.spring.boot.process;
+
+import org.activiti.api.process.model.builders.ProcessPayloadBuilder;
+import org.activiti.api.process.runtime.ProcessRuntime;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.test.context.support.WithUserDetails;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.IOException;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+@TestPropertySource(
+        locations = {"classpath:application.properties"}
+)
+@ContextConfiguration
+public class ProcessRuntimeConnectorTest {
+
+    private static final String CATEGORIZE_IMAGE_CONNECTORS_PROCESS = "categorizeProcessConnectors";
+
+    @Autowired
+    private ProcessRuntime processRuntime;
+
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+
+
+    /**
+     * It tests two connector actions inside the xml against two different connector json definitions:
+     * the first input variable is defined in the first connector action,
+     * the second input variable in the second connector action.
+     *
+     **/
+    @Test
+    @WithUserDetails(value = "salaboy", userDetailsServiceBeanName = "myUserDetailsService")
+    public void shouldConnectorMatchWithConnectorDefinition() throws IOException {
+        processRuntime.start(ProcessPayloadBuilder.start()
+                .withProcessDefinitionKey(CATEGORIZE_IMAGE_CONNECTORS_PROCESS)
+                .withVariable("input-variable-name-1",
+                        "input-variable-name-1")
+                .withVariable("input-variable-name-2",
+                        "input-variable-name-2")
+                .withVariable("expectedKey",
+                        true)
+                .build());
```

Comment:
May I suggest you to use a process containing a user task after the connector so that you can check the variable updates after the connector execution?

Reply:
mmm i am not sure i understand the domain here, can we add it as separate task after this?

Reply:
ok